src/Order/amendProductOnOrder.ts

```diff
@@ -1,47 +1,33 @@
 import { HttpModule } from '@whppt/api-express';
-import remove from 'lodash/remove';
 import assert from 'assert';
 import { Order } from './Models/Order';
 
 const addProduct: HttpModule<{ productId: string; quantity: number; orderId?: string | undefined }, Order> = {
   authorise({ $roles }, { user }) {
     return $roles.validate(user, []);
   },
-  exec({ $id, $mongo: { $dbPub, $saveToPubWithEvents, $startTransaction }, createEvent }, { productId, quantity, orderId }) {
+  exec({ $id, $database, createEvent }, { productId, quantity, orderId }) {
     assert(productId, 'Product Id is required.');
     assert(quantity, 'Product quantity is required.');
     const quantityAsNumber = Number(quantity);
-
-    return $dbPub
-      .collection('orders')
-      .findOne<Order>({ _id: orderId })
-      .then(loadedOrder => {
+    return $database.then(({ document, startTransaction }) => {
+      return document.query<Order>('orders', { filter: { _id: orderId } }).then(loadedOrder => {
         const order = {
           ...loadedOrder,
-          _id: (loadedOrder && loadedOrder._id) || $id(),
+          _id: (loadedOrder && loadedOrder._id) || $id.newId(),
           items: (loadedOrder && loadedOrder.items) || [],
         } as Order;
+
         const events = [] as any[];
+
         if (!loadedOrder || !loadedOrder._id) events.push(createEvent('CreatedOrder', order));
```

Comment:
Should this command be renamed to changeItemQuantity?
If it is changing the quantity then it shouldn't create an order